cpp/ql/lib/semmle/code/cpp/exprs/Expr.qll

```diff
@@ -1015,7 +1015,9 @@ class DeleteOrDeleteArrayExpr extends Expr, TDeleteOrDeleteArrayExpr {
   Expr getExpr() {
     // If there is a destructor call, the object being deleted is the qualifier
     // otherwise it is the third child.
-    result = this.getChild(3) or result = this.getDestructorCall().getQualifier()
+    if exists(this.getChild(3)) and not this.getChild(3) instanceof ReuseExpr
+    then result = this.getChild(3)
+    else result = this.getDestructorCall().getQualifier()
```

Comment:
This effectively keeps the old behaviour here. Some things might break if we don't.